### PR TITLE
Laika 1.0 support

### PR DESCRIPTION
**Pull request by rhpvorderman**
Laika now supports sbt 1.0 (#120). Since we heavily depend on Laika for our documentation and we want to use the SBT 1.0 plugin, this is my attempt to enable it for sbt-site.

EDIT 2018/01/11 :
In this pull request I have kept the old 0.13 laikaSiteplugin to keep backwards compatibility with sbt 0.13 and added a new plugin for sbt 1.0 and higher.
I also split the tests accordingly.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @rhpvorderman,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by rhpvorderman** [comment]
Might need some help with this
`The command "gem install jekyll:2.5.3 nanoc:3.8.0 asciidoctor --no-rdoc --no-ri" failed and exited with 1 during .`

Fail to see how this has anything to do with my contribution...

**Comment by rhpvorderman** [comment]
Okay. Almost there, but I will not be able to finish this pull request as there as a tiny problem that I can not solve.

The 0.13 `LaikaSitePlugin` publishes in `target/laikaSite` by default. Using the same settings the 1.x `LaikaSitePlugin` publishes in `target/docs/site`. With a minor tweak I managed it to publish in `target/laikaSite/site`. But I cannot get rid of this `site` subdirectory, no matter what I try. Can somebody please assist? It is the only thing that is still needed to support Laika for sbt 1.x and ensure backwards compatibility.

File: build.sbt [hunk 1]
```diff
@@ -42,7 +42,15 @@ libraryDependencies ++= {
         (scalaBinaryVersion in pluginCrossBuild).value
       )
     )
-  } else Nil
+  } else {
+    Seq(
+      Defaults.sbtPluginExtra(
+        "org.planet42" % "laika-sbt" % "0.7.5",
```
Comment (jonas) [inline build.sbt]: Does the latest version only support sbt 1.0?

File: src/main/scala-sbt-1.0/com/typesafe/sbt/site/laika/LaikaSitePlugin.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import Keys._
+import com.typesafe.sbt.site.SitePlugin
+import com.typesafe.sbt.site.util.SiteHelpers
+import com.typesafe.sbt.site.SitePlugin.autoImport.siteSubdirName
+import com.typesafe.sbt.site.SitePlugin.autoImport.makeSite
+import laika.sbt.LaikaPlugin.autoImport.{Laika, laikaHTML, laikaSite }
+import laika.sbt.LaikaPlugin
+import Path.relativeTo
+
+object LaikaSitePlugin extends AutoPlugin {
```
Comment (jonas) [inline src/main/scala-sbt-1.0/com/typesafe/sbt/site/laika/LaikaSitePlugin.scala]: Unless there are considerable differences I suggest we keep the same plugin code. WDYT?

File: build.sbt (hunk 1 above)
Comment (rhpvorderman) [reply build.sbt]: Yep. Support for 0.13 has been completely dropped.

File: src/main/scala-sbt-1.0/com/typesafe/sbt/site/laika/LaikaSitePlugin.scala (hunk 2 above)
Comment (rhpvorderman) [reply src/main/scala-sbt-1.0/com/typesafe/sbt/site/laika/LaikaSitePlugin.scala]: The plugin code is virtually the same. Only some variable names have been refactored to reflect the changes in the laika plugin itself.

There is just this thing with the `target/laikaSite/site` that I cannot fix on my own. See comment above.

**Comment by ffinfo** [comment]
👍

**Comment by rhpvorderman** [comment]
@jonas can somebody help with this `target/laikaSite/site` issue? The site should be in `target/laikaSite`. I tried to do this but I just can't get it to work. I get the feeling this should be simple and that I am missing something. It is the only thing that is keeping the tests from succeeding.

**Comment by jonas** [comment]
Closing in favour of #125

**Pull request closed without merging**